packages/grant-explorer/src/features/api/passport.ts

```diff
@@ -1,65 +1,102 @@
+import { datadogLogs } from "@datadog/browser-logs";
 import {
-  submitPassport,
   fetchPassport,
   PassportResponse,
+  PassportResponseSchema,
   PassportState,
+  submitPassport,
 } from "common";
 import { useEffect, useState } from "react";
 
 export { submitPassport, fetchPassport, PassportState };
 export type { PassportResponse };
 
-export function usePassport({ address }: { address: string }) {
-  const [passport, setPassport] = useState<PassportResponse | undefined>();
+export function usePassport({ address }: { address: string | undefined }) {
   const [, setError] = useState<Response | undefined>();
 
   const [passportState, setPassportState] = useState<PassportState>(
     PassportState.LOADING
   );
 
+  const [passportScore, setPassportScore] = useState<number>();
+  const [passportColor, setPassportColor] = useState<
+    "orange" | "green" | "yellow"
+  >();
+  const [donationImpact, setDonationImpact] = useState<number>(0);
+
   useEffect(() => {
     setPassportState(PassportState.LOADING);
 
     const PASSPORT_COMMUNITY_ID =
       process.env.REACT_APP_PASSPORT_API_COMMUNITY_ID;
+
     if (PASSPORT_COMMUNITY_ID === undefined) {
       throw new Error("passport community id not set");
     }
-    const PASSPORT_THRESHOLD = 0;
 
     if (address && PASSPORT_COMMUNITY_ID) {
       const callFetchPassport = async () => {
         const res = await fetchPassport(address, PASSPORT_COMMUNITY_ID);
+
+        if (!res) {
+          datadogLogs.logger.error(
+            `error: callFetchPassport - fetch failed`,
+            res
+          );
+          setPassportState(PassportState.ERROR);
+          return;
+        }
+
         if (res.ok) {
-          const json = await res.json();
+          const scoreResponse = PassportResponseSchema.parse(await res.json());
 
-          if (json.status === "PROCESSING") {
+          if (scoreResponse.status === "PROCESSING") {
             console.log("processing, calling again in 3000 ms");
             setTimeout(async () => {
               await callFetchPassport();
             }, 3000);
             return;
-          } else if (json.status === "ERROR") {
-            // due to error at passport end
-            setPassportState(PassportState.ERROR);
+          }
+
+          if (
+            !scoreResponse.score ||
+            !scoreResponse.evidence ||
```

Comment:
Is the plan to Make those Explicit via Boolean()? Because eslint complains when I do that, that it's supposedly unnecessary. Personally it's very clear what these are doing. I wish js had an idiomatic way to check for both null and undefined that wasn't `==`.

Reply:
I usually enable https://palantir.github.io/tslint/rules/strict-boolean-expressions/. 

It's primarily a matter of guarding against conditions greedier than expected. Maybe you've found yourself trying to figure why something like this would execute, when `s` is `string | undefined` and you've already verified that it's not `undefined`:

```ts
if (!s) {
  console.log('whooops')
}
```

Then you're at your wits' ends and finally decide to `console.log(s)`. Turns out that `s` was an empty string, which is falsy. Whooops.

(Same thing when `s` is a `number` and `0` starts triggering all your `null | undefined` conditions.)

I get the pain of writing `if (a === null | a === undefined)`. For cases like these I like to use `isPresent` from [ts-is-present](https://www.npmjs.com/package/ts-is-present).